### ISSUE TITLE
Caption support

### ISSUE THREAD

**Issue by jessetan** [carlos-jenkins/plantweb#6]
It is not possible to add a caption to a diagram. In Sphinx, this is done using the directive contents, but since the `uml` directive accepts uml as content, it would be better to use a `:caption:` option.

Implementation idea:
Make class `Plantweb` inherit from `Figure` instead of `Image` and set the value of `:caption:` as the `Figure` directive contents. May change current behavior since `Figure!=Image`.
Alternatively, just build and append a caption node in a way similar to `Figure` or sphinxcontrib-plantuml.

**Comment by compiaffe** [comment]
@jessetan Would this also allow for numerating uml diagrams like normal figures?

**Comment by jessetan** [comment]
If you're talking about Sphinx automatically adding "Figure XYZ" underneath a diagram, yes, that would happen if `PlantWeb` inherits from `Figure`

**Comment by compiaffe** [comment]
Yes that was my question. The plantuml plugin here: https://github.com/sphinx-contrib/plantuml
Does already add the `Figure XYZ` but requires a locally deployed plantuml.jar
